IPs/RSKIPxxx.md

```diff
@@ -0,0 +1,262 @@
+
+To obtain the ordFedAddress, the following bridge method must be called:
+
+`String getFederationOrdinalsAddress()`
+
+The address is constructed differently from the BTC peg-in address. The construction is described in the section “Main Ordinals Address Derivation”
+
+To register an ordinal, the following method can be called:
+
+`void registerBtcTransactionWithOrdinals(bytes rskTx, bytes btcTxSerialized, int height, bytes pmtSerialized)`
+
+This method will register the UTXOs as independent entities. Each UTXO registered will be assigned a serial number (or index) starting from 0 on each transaction.  This serial number matches the UTXO output index. If there are outputs that are not transfers to the federation ordinal address, then those serial numbers will be skipped.
+
+The UTXOs will be owned in RSK by the same private key that controls the first input of the transaction or owned by a different RSK address, if there is an `OP_RETURN` output containing it, as specified by RSKIP-170.
+
+
+## Peg-In using Destination-Mapped Federation Addresses
+
+It’s also possible to create a Bitcoin address that automatically assigns the ownership of the ordinal in Rootstock to a specific RSK address or contract. This is called destination-mapped federation addresses (DMFA). This can be used to automatically mint an NFT when the ordinal is transferred to Rootstock. 
+
+This requires creating a special unique single-use reception address for the ordinal. The address is created with the following method:
+
+
+`void registerBtcTransactionWithOrdinalsDMFA(
+   bytes rskTx,
+   bytes btcTxSerialized,
+   uint32 height,
+   bytes pmtSerialized,
+   address rskOwner);`
+
+The field rskOwner specifies the owner of the ordinal.
+
+## Changing the Ownership of an Ordinal
+
+To query who owns an ordinal, the following method is added:
+
+`address getOrdinalOwner(uint256 tId, int index)`
+
+The method will return 0x00... (the zero 20-byte address) if the ordinal does not exist.
+
+To change ownership of an ordinal, the following method must be called by the owner:
+
+`void transferOrdinal(uint256 tId, int index, address newOwner)`
+
+The method does nothing if the ordinal does not exist.
+
+## Ordinal Peg-outs
+
+To send an ordinal, the following method must be called:
+
+`void releaseOrdinal(uint256 txId, uint256 index, bytes btcAddress)`
+
+This method will queue the release of an ordinal, sending it to the btcAddress specified. The caller must be the owner of the ordinal. Every 1 hour the ordinals that are queued to be released will be processed. The bridge may attempt to create a single transaction containing all ordinal UTXOs, and all ordinal outputs, plus a single additional input used for fee payments, and a single additional output used for change.
+
+Ordinals can be sent to any Bitcoin address.
```

Comment:
The Bridge currently does not support creating transactions to segwit addresses. Should that be a pre-requisite or should we add a limitation to the type of recipient addresses?

Reply:
We can add the limitation. But I think that sending to any Bitcoin address would not be very complex to implement. It's just parsing the address and converting it to the correct scriptpub.